packages/synapse-react-client/src/components/TeamBadge.tsx

```diff
@@ -9,12 +13,28 @@ export type TeamBadgeProps = {
 }
 
 export default function TeamBadge(props: TeamBadgeProps) {
-  const { teamId, teamName, disableHref } = props
+  const { teamId } = props
+  let { teamName, disableHref } = props
+
+  let icon: IconName = 'team'
+
+  if (teamId == AUTHENTICATED_PRINCIPAL_ID) {
+    icon = 'public'
+    teamName = 'All registered Synapse users'
+    disableHref = true
+  }
+  if (teamId == PUBLIC_PRINCIPAL_ID) {
+    icon = 'public'
+    teamName = 'Anyone on the web'
+    disableHref = true
+  }
```

Comment:
Special cases for these teams--SynapseTableCell does this. In a future change, SynapseTableCell will use this component, so this brings parity.